### DISCUSSION TITLE
API:  Be able to get region from device

### DISCUSSION THREAD

**Discussion by ntaken** [netbox-community/netbox#11354] [Ideas]
**Proposed functionality:**

Using GET /api/dcim/devices/NUMBER/ to retrieve region and subregion that the device is part of.

**Use case:**

Being able to simply retrieve the entire(region + site + location) placement of a device with one API call. 

or

When using nornir_netbox for creating lists of devices based on their placement. 

For example I want to be able to script a script that sort devices into connection groups (for ssh) based on their placement.

**Comment by candlerb** [comment]
Basically, you're asking for more information about the Site to be returned.  The problem is that the more "related" information that is returned automatically, the more expensive each API call gets, as it has to join to more database tables.

Have you tried using GraphQL?
